src/datasets/commands/dummy_data.py

```diff
@@ -112,8 +112,9 @@ def _create_dummy_data(
         encoding = encoding or DEFAULT_ENCODING
         if os.path.isfile(src_path):
             logger.debug(f"Trying to generate dummy data file {dst_path}")
+            dst_path_extension = Path(dst_path).suffixes[0] if Path(dst_path).suffixes else ""
             line_by_line_extensions = [".txt", ".csv", ".jsonl", ".tsv"]
-            is_line_by_line_text_file = any(dst_path.endswith(extension) for extension in line_by_line_extensions)
+            is_line_by_line_text_file = dst_path_extension in line_by_line_extensions
```

Comment:
Not sure exactly how `Path` gets the suffixes, but this feels like it might fail for people who use dots in their file names (e.g. would `train.en.jsonl.tar.gs` give you `en` or 'jsonl'?)

Thoughts @albertvillanova? Any reasons not to check every suffix?

Reply:
@yjernite you are right: `train.en.jsonl.tar.gz` would give `en`, which is wrong. Therefore, I am implementing the check for every suffix.